src/module/macros/_hotbar.ts

```diff
@@ -0,0 +1,147 @@
+      actorId = orig_doc.actor?.id ?? "error";
+      itemId = data.id;
+    } else {
+      return;
+    }
+
+    switch (data.type) {
+      case EntryType.SKILL:
+        command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+        img = `systems/${game.system.id}/assets/icons/macro-icons/skill.svg`;
+        break;
+      case EntryType.TALENT:
+        command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}", {rank: ${data.rank}});`;
+        img = `systems/${game.system.id}/assets/icons/macro-icons/talent.svg`;
+        break;
+      case EntryType.CORE_BONUS:
+        img = `systems/${game.system.id}/assets/icons/macro-icons/corebonus.svg`;
+        break;
+      case EntryType.PILOT_GEAR:
+        command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+        img = `systems/${game.system.id}/assets/icons/macro-icons/generic_item.svg`;
+        break;
+      case EntryType.PILOT_WEAPON:
+      case EntryType.MECH_WEAPON:
+        command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+        img = `systems/${game.system.id}/assets/icons/macro-icons/mech_weapon.svg`;
+        break;
+      case EntryType.MECH_SYSTEM:
+        command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+        img = `systems/${game.system.id}/assets/icons/macro-icons/mech_system.svg`;
+        break;
+      case EntryType.NPC_FEATURE:
+        switch (item.FeatureType) {
+          case NpcFeatureType.Reaction:
+            command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+            img = `systems/${game.system.id}/assets/icons/macro-icons/reaction.svg`;
+            break;
+          case NpcFeatureType.System:
+            command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+            img = `systems/${game.system.id}/assets/icons/macro-icons/mech_system.svg`;
+            break;
+          case NpcFeatureType.Trait:
+            command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+            img = `systems/${game.system.id}/assets/icons/macro-icons/trait.svg`;
+            break;
+          case NpcFeatureType.Tech:
+            command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+            img = `systems/${game.system.id}/assets/icons/macro-icons/tech_quick.svg`;
+            break;
+          case NpcFeatureType.Weapon:
+            command = `game.lancer.prepareItemMacro("${actorId}", "${itemId}");`;
+            img = `systems/${game.system.id}/assets/icons/macro-icons/mech_weapon.svg`;
+            break;
+        }
+        break;
+    }
+
+    // TODO: Figure out if I am really going down this route and, if so, switch to a switch
+    if (data.type === "actor") {
+      title = data.title;
+    } else if (data.type === "pilot_weapon") {
+      // Talent are the only ones (I think??) that we need to name specially
+      if (data.type === EntryType.TALENT) {
```

Comment:
Note for future - this is weird. Why are talents inside pilot weapons? Seems unreachable.

Reply:
I removed this entire branch system in #375